### ISSUE TITLE
New governance model

### ISSUE THREAD

**Issue by ploeh** [AutoFixture/AutoFixture#703]
AutoFixture needs a new governance model. From it's humble [beginnings in 2009](http://blog.ploeh.dk/2009/03/22/AnnouncingAutoFixture), I've been using the [benevolent dictator governance model](http://www.outercurve.org/resources/governance/benevolent_dictator.html), and for many years, this has worked well (in my opinion).

It's always been my responsibility to steer AutoFixture in the direction of my vision. For the last couple of years, however, I've lost the vision for AutoFixture. In my opinion, it's _done_. I don't need it to do more than it already does.

This is exacerbated by the fact that I rarely write C# code anymore. While I could use AutoFixture in F#, I prefer [FsCheck](https://fscheck.github.io/FsCheck) instead.

Running the AutoFixture project has always been a labour of love for me. Increasingly, however, I find that I've lost the passion. Recent contributors have probably felt this, and for that, I apologise.

It's time to try something else in order to enable the project to move forward. I can think of at least three options, but I'm open to more:
- Continue with the _benevolent dictator_ governance model, but with a new leader. This person must be someone willing to take on the responsibility associated with that role, as well as be someone the AutoFixture community trusts with that role. This could be one of the current core contributors to AutoFixture (@adamchester, @ecampidoglio, @moodmosaic), but could also be someone coming in from the outside.
- Establish a new governance model. It'll be up to the people taking over to decide which one.
- Establish a sponsorship. While I've lost the passion for AutoFixture, I still _care_ about it, and would be willing to continue if compensated. Managing AutoFixture has never been a full-time job, so I don't expect a big sum; frankly, I don't expect this option to be realistic, but if I don't ask, the answer is automatically _no_ :wink: 

Obviously, I'll be happy to assist handing over the responsibilities of the project to someone else. You may, for example, wonder how to build, sign, and release new NuGet packages. It's easy, but you'll need the signing key.

It's been great leading AutoFixture for all those years. It's a project that I originally created and nurtured. It's difficult to let go :cry: 

My hope is that this is like a child growing up, now becoming mature enough to move on without his or her parents.

**Comment by moodmosaic** [comment]
I would lean towards the third option; *Establish a sponsorship*, hoping to keep AutoFixture at the same *fine* quality it currently is.

**Comment by Mykezero** [comment]
@ploeh Thank you for all the hard work and dedication you've put into this project! Of course, I would always be willing to sponsor you by buying you those cups of coffee ^^

In my opinion, the benevolent dictator model could work if one of the core contributors has the following: 

1. A future vision for the project with a direction of where AutoFixture is heading towards. 
2. The passion for the project along with a long term commitment to its community.
3. The technical understanding of AutoFixture as a whole.
4. A willingness to do that work that must be done to drive toward that future vision. 

If that's not the case, then a new model where the core contributors are in charge of making the decisions for what should and shouldn't be done in the project. This would require a shared vision for the future of AutoFixture, and a lot of communication to discuss new changes within the project. 

Hopefully I've hit on a few key notes with regards to a new governance model, and that my comments aren't totally out of sync with the realities of running an open source project.

**Comment by moodmosaic** [comment]
I've been around and helping on `2.` and `3.`, on and off since mid-2011, and I'd be happy to continue doing so.

**Comment by ecampidoglio** [comment]
As much as it pains me to see this era of AutoFixture come to an end, I fully understand @ploeh's motivations for stepping down as the project's leader. Thank you for the time and effort that you've invested in AutoFixture, Mark! Not to mention the knowledge that you've shared in your code reviews — I've learned tons from those!

For me AutoFixture remains a tool that I use almost on a daily basis and I'd like to see it continue to improve. Like @moodmosaic, I've also been contributing — albeit to a lesser extent — since 2011 and I certainly intend to stay committed.

Having said that, I must admit that I don't feel I have the necessary vision to lead the project forward. Sure, I have a few ideas; for example, it would be nice to have a smaller and more straightforward API (right now there are multiple ways of achieving the same thing, with various degrees of discoverability). But that's far from being a well-defined goal. More like a general direction, at best.

So here's my proposal: we drop the individual leader model and instead offload those responsibilities to the @AutoFixture/core team as a whole. I think the group is small and experienced enough to be able to handle future design decisions together.

What do you guys think?

**Comment by sergeyshushlyapin** [comment]
I can only say thank you @ploeh, @moodmosaic, @ecampidoglio and @adamchester for reviewing my pull requests! Usually it was quite challenging :wink:, but I appreciate your input and continuous knowledge sharing!

I really enjoyed contributing to AutoFixture and "I Don't Wanna Stop" (c). I do believe the core team will continue doing their excellent job. Well done, guys!

**Comment by bjorn-ali-goransson** [comment]
No vision is better than a bad vision. Which is why I'd (spontaneously) prefer someone similar to @ecampidoglio to lead the project, even if (if) this would mean that it almost goes into maintenance mode.

("Promote a leader who above all would not wish leadership" as we muslims say)

Furthermore, if AF turns into something else because of the new dictator´s vision, then the original AF would soon again be without dictator. So, let's value a relative status quo for a while, to consolidate.

Not to show disrespect your efforts, Mark, but let's not forget the old question of the ploeh namespace prefix, while you're still onboard.

**Comment by ploeh** [comment]
Revisiting this thread three months later, it looks like AutoFixture is stuck. I did realise, however, that if anyone wants to take over the reins, they'd need at least two things from me: the signing key (`.snk`) and publish rights to the NuGet packages. With these, I believe, someone else should be able to publish new versions of AutoFixture.

Contact me if you'd like to have these transferred to you. Do understand, however, that I will regard any such transfer as an official transfer of the rights, as well as the responsibilities, of future maintenance of AutoFixture.

**Comment by sergeyshushlyapin** [comment]
To keep balance between @ploeh's desire to depart away from AutoFixture and the community desire to get new versions (see amount of links to another PBIs in this thread), I would like to suggest the following workflow:

1. Core contributors (@ecampidoglio, @moodmosaic or @adamchester) review and discuss coming pull requests.
2. As soon as they are done with review and a PR is ready to be merged, they signal to @ploeh.
3. @ploeh in any suitable for him time simply pulls the trigger and release a new version _without_ getting into release details.

Wouldn't it satisfy everyone's needs?

**Comment by moodmosaic** [comment]
> 1. Core contributors (@ecampidoglio, @moodmosaic or @adamchester) review and discuss coming pull requests.

Fine by me, *if* I have the required knowledge to review the PR(s). Usually, I (happily) review parts of the code-base where I have previously worked-on, created, or contributed.

**Comment by ploeh** [comment]
Since I added this issue months ago, few people have noticed it. Most people still think that I'm the benevolent dictator of AutoFixture.

The model suggested by @sergeyshushlyapin will only confuse matters. It has to be clear to contributors what the governance model is, and who's making decisions.

The only thing required of me in that suggested model is that I perform some routine operations. It's nothing more than 5-10 steps in the Git Bash, and I'll be happy to teach someone else to go through those motions instead of me. There's no reason it has to be me, other than currently, as explained [above](#issuecomment-265129971), I'm the only person in possesion of the keys to the resources.

The model suggested will, de facto, put the responsibility on @moodmosaic, @ecampidoglio, and @adamchester. If they are fine with that, they should also take on the formal responsibility. All of them could get my keys, and thereby sharing the responsibility, but I'm not interested in retaining formal responsibility (unless a sponsorship pops up).

**Comment by adamralph** [comment]
Why is the signing key kept secret?

**Comment by ploeh** [comment]
What's the point of signing anything if the signature key isn't secret?

**Comment by adamralph** [comment]
I can see where this is leading. I now ask:

> What's the point of signing anything?

And hey presto! We're into a signing/not-signing battle. 😉

I don't want to hijack this issue. That war is being waged elsewhere. Sorry for mentioning it.

**Comment by ploeh** [comment]
> We're into a signing/not-signing battle.

That wasn't really my intention; sorry. I don't have a hard opinion about that debate. To be frank, I still can't fully get my head around it, because there's so much wrong information being thrown around between the correct information... I have a hard time separating the signal from the noise, so I tend to favour keeping things that have been working for years as they are...

AutoFixture is fairly old, and it's always been signed by that key. Microsoft strongly recommended it back then..

**Comment by adamralph** [comment]
I wouldn't un-sign it now. The problem with that is that the assembly identity would change, so things like binding redirects would no longer work. I'm also looking after some projects which were signed back in the day and I've resigned myself to the fact that I'll just have to keep them signed forever. Oh, and there's also the problem of virality. I.e. another reason I can't un-sign those projects is that there may be signed project out there that depend on them.

OK, one last word on the secret key: have you considered what you would do if the key became compromised? Changing the key would also change the assembly identity. (log4net actually made the decision to do this in order to make their key secret, and it caused no end of problems.)

**Comment by ploeh** [comment]
Good point about the key becoming compromised. It looks like it's a moot point for AutoFixture now, but if it ever happened, my first reaction would be to exchange the key with a new one. Since that changes the identity of the assembly, we'd have to increment the major version.

I suppose another alternative at that point would be to stop signing the assembly altogether, but that would still require a new major version.

**Comment by adamralph** [comment]
That's what log4net did, but it didn't work out well. It caused considerable pain and confusion for consuming apps and dependent packages. (I gave up and dropped log4net.) I don't know how much of a problem it would be for AutoFac though.

Since the identity of the assembly changes, it's not a new entry in the version history of the assembly. It's the start of the version history for a _new assembly_. If a key must be secret, but becomes compromised, then it's not possible to release a new version of the assembly.

**Comment by DerAlbertCom** [comment]
so, no one of  @ecampidoglio, @moodmosaic or @adamchester seems willing to take the responsibility. So I propose @Kralizek which done the port to .NET Core. For me AutoFixture seems feature complete. So i clear lack a Vision here. For me it is enought that it works on the current available .NET Framework. A .NET Core Port is needed, they are two ways, a fork or a friendly change of the ownership. I'm all in for a friendly change, but I would also use the fork.

**Comment by wallymathieu** [comment]
@DerAlbertCom sounds like both @ecampidoglio and @moodmosaic are OK with having shared responsibility...

**Comment by Kralizek** [comment]
@DerAlbertCom thanks for naming me, but I simply pushed a PR. Glad to help, but other people know the project more than I do.

**Comment by Zvirja** [comment]
Hi,

Unfortunately, it appeared that again we see the situation when everything stuck. It's very bad, because there are changes that definitely should be merged to not make this project dead (like .Net Core support).

Therefore, I'd suggest the following plan. Of course, if somebody could offer something better - do that. I'm just trying to push from the dead point.

1. @ploeh will create the `@AutoFixture/core` team as it was suggested by @ecampidoglio [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-246491296). The team members for now are our good old @ecampidoglio, @moodmosaic and @adamchester. This team should have admin rights for the company account (if that is possible) and the AutoFixture repository itself.

2. @ploeh writes a letter to all 3 guys (or somebody one, so he could share the data later) and provides with the necessary keys (signing key, NuGet key, etc) and all the necessary instructions (e.g. how to push the new release).

By doing these 2 steps Mark enable guys from core team from being able to merge PRs and release new versions of AutoFixture. Also this would transfer the ownership :)

__Subsequent steps:__
After guys obtain permissions, it will become possible to review PRs and publish new versions. Later we could expand our Core team if guys are too busy to review PRs or lost the interest.

Also it might make sense to review when new versions of AutoFixture are published. I'd suggest to push to NuGet on each push to the master. We could review tools for auto-tagging or have some mark inside the source code - but that would be discussed later.

@ploeh, @moodmosaic, @adamchester, @ecampidoglio - do you agree with this plan? If so, let's start to do it - nothing better was suggested before and likely will be suggested :)

P.S. I'm ready to contribute and volunteer if there is something I can help with (e.g. configure CI) - I like AF and want it to be alive.

**Comment by ploeh** [comment]
@Zvirja thank you for writing, and for your concern. Most of what you suggest is already in place. @ecampidoglio, @moodmosaic, and @adamchester already have commit rights to the AutoFixture repository. I'm also fairly sure that at least @ecampidoglio has admin rights in the AutoFixture GitHub organisation, because, IIRC, he originally created it.

We've been doing [continuous delivery with AutoFixture since 2011](http://blog.ploeh.dk/2011/09/06/AutoFixturegoesContinuousDeliverywithSemanticVersioning).

The only thing that new maintainer(s) need in order to take over are:

* Commit rights (already in place for the core members)
* Strong name keys
* NuGet publication rights

As soon as someone publicly request all three, I'll be happy to transfer all of them. Once this is done, I'll consider maintenance responsibility to have officially transferred.

I'm only waiting for someone to take on the responsibility. There are no blockers apart from that.

**Comment by Zvirja** [comment]
@ploeh Thanks for the reply, now situation is much clear! It appears that this team is private as I don't see it [here](https://github.com/AutoFixture). Could we make it public, so it's visible?

The next step would be to clarify the readiness and concerns of the Core team members. I see that some of the guys replied previously (except @adamchester). Therefore, I'd like to ask @adamchester, @moodmosaic and @ecampidoglio to answer here explicitly:

1. Do you agree to be one of the keys holder? It doesn't seem that this put some extra responsibilities, except the fact that you should keep them in safe place and probably be ready to share them to new Core team members (if such thing happens in future).

1. Are you still interest in reviewing AutoFixture Pull Requests (if you have free time and appropriate knowledge in particular domain, of course)?

1. Are you ready to manage the releases when needed? Meaning that after PR is merged, you trigger the push procedure increasing the version in accordance with SemVer we follow here. It doesn't seem to be time-consuming stuff.

1. Do you have special concerns regarding your membership in the AutoFixture project life? (That could be e.g. reluctance to do take particular responsibilities non-mentioned above, luck of time for helping often, etc).

By answering this question we as community will clearly understand your position and role, so we could see what we have and whether we need somebody else to make this project alive.

After this reply we will see whether what are the next steps. For instance, we could involve new members like @klimisa suggested above.

**Comment by adamchester** [comment]
I'll continue to help out where I can, but I cannot take responsibility for the project or releases unfortunately.

**Comment by moodmosaic** [comment]
>1. Do you agree to be one of the keys holder? It doesn't seem that this put some extra responsibilities, except the fact that you should keep them in safe place and probably be ready to share them to new Core team members (if such thing happens in future).

I'd prefer to not hold the keys if I'm not going to be involved with the release cycle.

>2. Are you still interest in reviewing AutoFixture Pull Requests (if you have free time and appropriate knowledge in particular domain, of course)?

Absolutely, yes. Like I mentioned [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-246267511), I've been doing this on and off since mid-2011. Happy to continue doing so. However

* I'm pretty busy this period and so there might be delays if I have to review pull request(s)
* As I mentioned [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-275347457), I usually review parts of the code-base where I have previously worked on, created, or contributed *(that is, I don't hold strong opinions about supporting .NET Standard, for example)*

>3. Are you ready to manage the releases when needed? Meaning that after PR is merged, you trigger the push procedure increasing the version in accordance with SemVer we follow here. It doesn't seem to be time-consuming stuff.

I'd prefer to stay out of the release cycle, but I do hope that @klimisa, or someone else, will take care of this. It should be a fairly simple process, like @ploeh wrote [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-275415331).

>4. Do you have special concerns regarding your membership in the AutoFixture project life? (That could be e.g. reluctance to do take particular responsibilities non-mentioned above, luck of time for helping often, etc).

No special concerns, but—please—pay close attention to what I wrote in `2`.

HTH :wink:

**Comment by klimisa** [comment]
As I already said [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-289997920), I'm very keen to take over the releases. What I understand from the previous comments is that there is no other blockers other than someone to ask for it publicly.

Therefore I kindly ask @ploeh to transfer to me the things that mentioned [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-289989346), which are necessary in order to take over.

@ploeh how you would like to proceed with the transfer?

**Comment by Zvirja** [comment]
@moodmosaic and @adamchester, thank you for the replies.
I'd suggest to wait for the @ecampidoglio reply as well before we move forward.

BTW, @klimisa, could you clarify your background with AutoFixture? Your name looks pretty new - I wasn't able to find any issues or Pull Requests from you. I'd vote to grand permissions to somebody with background if possible, as this access is pretty sensitive.

**Comment by ecampidoglio** [comment]
First and foremost, I'd like to thank @klimisa for stepping up and volunteering to take over the releases of AutoFixture. Like the rest of the core team, I, too, don't have the bandwidth to take on that responsibility at this point in time.

However, I _am_ willing to review PRs, especially the ones that touch portions of the code I've worked on myself or whenever I feel I can offer some input.

Speaking of which, I suggest that, even with @klimisa as the _release manager_, all PRs should be approved by at least two people from the @AutoFixture/core team before being merged into `master` and, subsequently, released.

**Comment by Zvirja** [comment]
Thanks, now we have all the replies! That is perfect.

So we have the situation when all the Core team members are glad to review PRs, while nobody wants to take care of the Release Management. Therefore, the next step is to find the "key keeper".

As I mentioned above, I'd vote to provide keys only to somebody with history in this project. I wasn't able to find any history entries related to @klimisa - let me know if I blindly missed something.

In the meanwhile, I've checked the commit stats. Here they are:
-  1779  Mark Seemann (@ploeh)
-   524  Nikos Baxevanis (@moodmosaic)
-   129  Enrico Campidoglio (@ecampidoglio)
-   105  Gert Jansen van Rensburg (@gertjvr)
-    93  Adam Chester (@adamchester)
-    71  Hackle Wayne
-    66  Marcio Rinaldi (@mrinaldi)
-    54  Diogo Castro
-    42  Sergey Shushlyapin (@sergeyshushlyapin)
-    .....
-    9 Alex Povar (@zvirja)


I've checked the recent contributors list and found a spike of activity from @sergeyshushlyapin. Also, he is one of the top by commits (and contributed 25 PRs). It looks like a good candidate. @sergeyshushlyapin what do you think about this?

Also, I'd like to offer myself for the Release Management - I contributed a bit to this project, I'm monitoring its issues from time to time and I'm actively using AutoFixture in my work. I have about 1 year of AF experience. Also, I spent [a lot of time](https://github.com/nsubstitute/NSubstitute/pull/259) extending NSubstitute to [better integrate with AF](https://github.com/AutoFixture/AutoFixture/pull/726), so it could prove my interest.
Of course, I'd be happy to review PRs as well, but I'm not sure that I have enough knowledge about AF.

I'd suggest the following further steps:
1. Wait for the @sergeyshushlyapin reply.
2. Vote for the candidates afterwards.
3. Share keys to the winner(s) and decide upcoming plans.

Let me know if somebody have different vision or disagree with my suggestions (or I am unaware of something important).

**Comment by ploeh** [comment]
Thank you, @klimisa. I've sent you an email in order to start the hand-over.

**Comment by Zvirja** [comment]
@ploeh Could we please postpone this for a bit? I'm a bit concerned about such a decision.

As I mentioned above, the keys should be trasfered so someboby involved to the project, somebody we know and could trust making the keys safe. This is pretty sensitive information, given that AutoFixture is widely used package. Sharing keys with unknown people could lead to unpredictable results - starting from undesirable package management and ending with upload of malicious packages to NuGet (just to highlight the importance).

As I mentioned above, I wasn't able to find any history related entries about @klimisa. Do you know him personally? We should transfer this data only to somebody we know - so we could trust.

Sorry for my concerns, but I use this project on the work, so I want to trust the uploaded packages and be sure that there will be no issues with project itself.

That is why I suggested @sergeyshushlyapin or myself - I have some background which could prove my clear intentions.

**Comment by ploeh** [comment]
@Zvirja, yes, [I've met](https://twitter.com/ploeh/status/844508365543931904) @klimisa. He's an acquaintance of @moodmosaic.

This issue has been open for more than half a year, so it's not that people are standing on each other in order to take over the reins. Until recently, it looked like the choice was between no one and anyone.

I completely understand your concern (although, having met @klimisa, I suspect no ill intent), but until a moment ago, I hadn't noticed any other volunteers. If you volunteered yourself before now, then please accept my apologies for missing that.

As I originally wrote, I want nothing else for AutoFixture than see it continue thriving. Until two days ago, it looked dead, so I found it rational to take the chance offered.

Now there are two volunteers. The key holder doesn't have to be a single person, but I do understand the concern.

Since more than half a year has gone by, there's no reason to rush into this, so I'm happy to wait a couple of days to give anyone with an interest a chance to chime in.

**Comment by Zvirja** [comment]
@ploeh Thanks for the reply and for your understanding of my concerns. It isn't very comfortable to write such things and look like you suspect people around, but I haven't met @klimisa somewhere before, so felt that it's better to re-insure.

Yes, I suggested myself in my [previous reply](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-290228042). It will be a proud for me to participate in this project. But I want to see whether there are volunteers with larger background. That's why I suggested @sergeyshushlyapin. Let's wait for his reply. After that we will likely have full list of candidates.

I do understand that there was a silence here, but I believe that this time it will be decided. As you can see, we already know commitments of all the core team members, so it's much easier now.

**Comment by sergeyshushlyapin** [comment]
@Zvirja, thank you for promoting me for this position and thank you for bringing attention to this thread. You've definitely succeeded with the last one!

From the beginning of this transition period, I believed that one of the core contributors should take the ownership over AF. As @ecampidoglio, @moodmosaic and @adamchester clearly stated few times, nobody of them is willing to do so and that is totally fine.

From my point of view, if two key AF contributors knows @klimisa in personal and ready to give him access rights, they should do that.

**Comment by Zvirja** [comment]
Thanks for the reply!

Fine, now we have examined all the obvious paths :)

So now we have only 2 candidates volunteered for the keys keeping and help with releases - @klimisa and me. I have AF contribution background, @ploeh and @moodmosaic know @klimisa personally. It's unlikely that someone will join in the nearest future. If it happens so - keys should be additionally shared.

If nobody minds, I'd like to be the key keeper and I'm ready to take the Release Management responsibility. As I stated before, I work with AF often and I have some contribution experience here - so I should be fine. If @klimisa is fine to work together as a release manager, let's make @klimisa the release manager as well. Doing so we will have 2 people responsible for releases. Hope we will synchronize fine and will keep the same direction (e.g. regarding the v4.0 release).

So let's wait for the @klimisa final word and then share the keys with me and @klimisa.

**Comment by klimisa** [comment]
For me, it doesn't matter how many people will hold the key, as long as there will be no conflicts.

I suggest in order to avoid conflicts, 
for every merge(s) a _release manager_ does, it should be followed with a Nuget.org publish by the same person.

**Comment by zvirja** [comment]
Given the replies above, @ploeh could you please start the keys and instructions transfer to both @zvirja and @klimisa?

**Comment by moodmosaic** [comment]
I don't see why not having both @zvirja and @klimisa on board for doing the releases, as long as they don't step on each other's toes:

1. as @ecampidoglio [wrote](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-290217987):
    >all PRs should be approved by at least two people from the @AutoFixture/core team before being merged into master and, subsequently, released.
2. as @klimisa [wrote](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-290784653):
    >for every merge(s) a release manager does, it should be followed with a Nuget.org publish by the same person.

**Comment by zvirja** [comment]
Yes, it should work, so let's wait for the @ploeh share. 

Later after we get keys, I'll check how everything is organized. I gave idea to simplify the release flow so, that all you need to do is just to create a Release in GitHub. That will push a tag and later CI will perform all the necessary work with build and push. Also we need a feed for pre-release builds, e.g. for v4 branch as it will take some time before we stabilize projects.

But the further steps will become clear only after I see how everything is configured currently :)

**Comment by ploeh** [comment]
@zvirja, could you please send me an email? Contact information here: http://blog.ploeh.dk/about

**Comment by zvirja** [comment]
@ploeh Sure, I've just sent you an email.

**Comment by ploeh** [comment]
The strong name key has now been handed over to @zvirja and @klimisa. They have also both been added as package owners on nuget.org.

As far as I'm concerned, this completes the transfer of AutoFixture. @zvirja and @klimisa are now the new maintainers. Congratulations, and thank you for taking on the responsibility 🎉 🎈 🍸 🍾 

This also means that I'm going to close this issue. Comments can still be added here, even when the issue has been closed, so this isn't a ban on further comments.

Unless you want to kick me off the project, I'll be happy to help by answering any questions you may have. Just open a new issue and mention me, or write here.

**Comment by jongalloway** [comment]
@zvirja @klimisa While you're working on changing over governance, copyrights, etc., are you interested in joining .NET Foundation?

**Comment by zvirja** [comment]
@jongalloway Well, I'd say that likely we haven't thought about this before :) While I've heard about .NET Foundation before and know that some of the popular projects are already the members (like xunit and BenchmarkDotNet), I'm not sure that I clearly understand what does that actually means.

Is that possible for you to provide us with more details on this so we could discuss that? In particular the following points are unclear:
- What does this mean at all for the AutoFixture as a project? :)
- What are the changes the project should apply?
- Which benefits/limitation this brings on?
- What is the procedure of taking the membership?

I'd appreciate if you have answers to this questions or could just point out to the right doc. It sounds as something cool :)

**Comment by cokkiy** [comment]
.NET Foundation has a FAQ here https://dotnetfoundation.org/about/TSG

**Comment by arangas** [comment]
There is also a .NET Rocks podcast show on the Foundation here: https://dotnetrocks.com/default.aspx?ShowNum=1217

**Comment by ploeh** [comment]
For everyone's information: I've unwatched the AutoFixture repository. If you need my input, you'll have to explicitly mention me in a discussion.

**Comment by moodmosaic** [comment]
Perhaps, we can revise [this](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-290921802) so that we allow a PR to be merged if it's a approved by *at least one* member of the AutoFixture core team?

I might be wrong, I haven't looked closely, but IIRC there are a few PRs by @zvirja that *do* have an approval, and still remain open for weeks (maybe a month or so?) because they're waiting the second approval.

**Comment by zvirja** [comment]
@moodmosaic Thanks for raising this question 👍 Yes, it's definitely a problem for now (I have 4 such PRs), as we cannot move on until the @ecampidoglio is missing (as @adamchester doesn't have too much time to review PRs) 😞 

I'd definitely vote to change the rule so that _at least one_ approval is required. Later particular reviewer could apply the local override saying smth like "Looks fine to me, but let's see what [other member] think" if he isn't confident.

@adamchester @ecampidoglio @klimisa What do you think?

**Comment by adamchester** [comment]
Agreed regarding at least one approval.

Regards,
Adam Chester

**Comment by zvirja** [comment]
To avoid potential issues, I'd suggest us to adjust a rule a bit. 

I propose that a PR could be merged only if it _either_:
- it has two approvals;
- it has a single approval and at least a week of lifetime.

With this approach we will ensure that other team members had a chance to review a PR and leave their comments. If they didn't appeared in a week - it's OK to proceed to avoid zombie PRs :)

If you are fine, I'll copy this rule to our [wiki](https://github.com/AutoFixture/AutoFixture/wiki/Project-processes).

**Comment by zvirja** [comment]
To avoid potential issues, I'd suggest us to adjust a rule a bit. 

I propose that a PR could be merged only if _either_:
- it has two approvals;
- it has a single approval, at least a week of lifetime and no pending reviews.

With this approach we will ensure that other team members had a chance to review a PR and leave their comments. If they didn't appeared in a week - it's OK to proceed to avoid zombie PRs :)

If you are fine with this change, I'll copy this rule to our [wiki](https://github.com/AutoFixture/AutoFixture/wiki/Project-processes).

**Comment by zvirja** [comment]
@moodmosaic @adamchester @ecampidoglio I've added the aforementioned rule the [project processes](https://github.com/AutoFixture/AutoFixture/wiki/Project-processes#pull-request-review) page of our wiki. If you disagree with text - let's discuss the changes.

P.S. I've already merged a couple of PRs today as majority agreed to follow a new rule.

**Comment by zvirja** [comment]
@klimisa @ecampidoglio Could you please find a couple of minutes and vote whether you are fine with [this](https://github.com/AutoFixture/AutoFixture/wiki/Project-processes#pull-request-review) project rule we want to introduce? That will enable project to move on as currently we are stuck 😞

Thank you!

**Comment by klimisa** [comment]
I'm ok in with new project rule. The only dispute I have is the one week lifetime. I suggest being more days, maybe 10 or 2 weeks. One week it's too short when being busy.

**Comment by zvirja** [comment]
@klimisa Thanks for the reply. Well, I had an idea that in one week you could check new PRs and estimate whether you have knowledge/time/intention to review them in the nearby future. If so, you can say about that and PR will wait for your review. 

On other hand, if you are completely missing, you will not block the project. 

@moodmosaic @adamchester What do you think about the lifetime? Do you agree that one week should be enough to review PRs and say that you will participate in review, or you also vote to increase the "timeout"?

**Comment by adamchester** [comment]
I don't want to block things here while I'm too busy, so I agree with @zvirja.

One week should be enough to review or drop a note that you intend to review in the near future.

**Comment by moodmosaic** [comment]
Yeah, I think 7 days should be a reasonable amount of time.

**Comment by zvirja** [comment]
I'm fine with 7 days as well. @klimisa Do you agree to proceed with one week?

**Comment by klimisa** [comment]
@zvirja that's fine with me. 
I suggested expanding the PRs lifecycle into 2 weeks for the reasons below,

- it gives more room in terms of time for more people to review a PR.
- It gives a reasonable time for a PR to mature.
- I don't find the 2 weeks time window as blocking

Anyhow one week should be Ok. 👍

**Comment by zvirja** [comment]
Let's proceed with one week and see how it goes. If later we see that it's not enough - we can always tune the rule and increase the lifetime.

@ecampidoglio It's only your voice that is missing now :wink:

**Comment by zvirja** [comment]
Would like to agree how we work with the Reviewers field in PRs, because found that currently I automatically assign @moodmosaic @adamchester and @ecampidoglio when I create a new PR 😄 Likely, that is not very effective as it doesn't reflect the real state of things.

I'd suggest to not assign nobody to PRs at the beginning. Later, if somebody decides to perform a review, he should self-assign himself. This will reflect that somebody is working on the review and that PR should not be merged until he doesn't approve/decline. If we want to invite somebody to PR as a reviewer, we could mention him in the discussion so later he will self-assign himself and perform the review.

Also that means that maintainers should review new PRs even if they are not assigned 😉 

Probably, that is the common practice for OSS project - I simply don't have too much experience :) @AutoFixture/core Do you agree on that? If so, I could post that to wiki.

**Comment by moodmosaic** [comment]
>Later, if somebody decides to perform a review, he should self-assign himself.

This may not work for me. I'm so busy from now on that I'm pretty sure I'll never go back to have a look and self-assign myself.

Thus, the only way to have me review a pull request is being explicit, so I can then take a note of the pull request `#` and get back to it eventually.

**Comment by zvirja** [comment]
@moodmosaic Oh, I see :) In this case I'll continue to follow the existing approach.

**Comment by zvirja** [comment]
Given that everybody except @ecampidoglio voted for the rule above about 1 vote + 1 week, I'll consider this rule agreed. Enrico, if you have any concerns - please let us know.

**Comment by zvirja** [comment]
Copying the conversation from [here](773#issuecomment-322139890) to keep organizational questions in the single place. 

@moodmosaic wrote:
> On a side note, I have a gut feeling that #703 must be revised, and move to a new model, probably similar to the previous one we had but this time with @zvirja on this position.

The background for this issue was the PR when we didn't have available reviewers with expertise in the .NET Core area, so we looked for the ways to move on.

Personally, I feel issues with the existing model, because it assumes that all the maintainers are active. The reality is that we have only me, @moodmosaic and partially @adamchester 😟 When I create any PR, I cannot self-approve it, so I need a review from either @moodmosaic or @adamchester. After we accepted the rule of a single approval and week of delay, the things become significantly better - I managed to merge a set of PRs 😃 However, we still have situations when we don't have available reviewers with area knowledge for long period of time (like for [vNext dev feed](https://github.com/AutoFixture/AutoFixture/pull/762)) and we don't have any ways to proceed. Of course, we can always refer to the [speed kills](http://programmer.97things.oreilly.com/wiki/index.php/Speed_Kills) article saying that it's fine to move sloooowly, however personally I feel more that we got stuck as no active discussions are present.

Personally, I could offer a new rule to solve the issue. It's quite fuzzy, so let's discuss your concerns and vision. I suggest if PR is fired by @AutoFixture/core member (whom we trust) and we don't have anybody with expertise available, then it's fine to merge the code. Of course, this should be used fairly:
- at least one week of lifetime is required;
- reviewer(s) with expertise should be explicitly invited;
- reviewer(s) without area knowledge should also review PR if possible to write about their concerns;
- no pending reviews;
- PR author should be confident in what he is suggesting.

With such approach if reviewer with domain knowledge is completely busy for long period of time, the project will not be blocked when non-breaking features/fixes are suggested. Of course, if something fundamental changes (e.g. architecture, massive redesign), it should be agreed with all the members.

My point is that even if something non-optimal is suggested and merged, that is not a very big problem, as we could always improve that via the subsequent PRs. Our code is greatly covered by tests, so in most cases we can trust the changes.

----

@moodmosaic @adamchester @klimisa Could you please share whether you see other ways to solve the issue? And if not, do you agree with the suggested rule?

Will be happy to know your concerns and vision, so we will be better aligned 😉

**Comment by moodmosaic** [comment]
Yes, it looks like @zvirja is excited and does a lot of (great) work, which is a good thing IMO. However, 
* @AutoFixture/core must able to follow and approve all that work, because *if* @zvirja looses all this excitement, someone else must be able to follow and maintain the project.

And here's the *[catch-22](https://en.wikipedia.org/wiki/Catch-22_(logic))*:

We could probably switch back to the [old model](http://oss-watch.ac.uk/resources/benevolentdictatorgovernancemodel), this time with @zvirja, however,
* *(read the very first bullet)*

---

Given that, I think the easiest is to continue with what it's agreed already. And it is my guess that, eventually,
* @adamchester and @ecampidoglio will show up and help finish the pull request reviews
* @klimisa might be able to be of some help, too 

HTH

**Comment by zvirja** [comment]
@moodmosaic Thanks for your opinion! 👍

Well, as I mentioned above, the existing model works, therefore we could try to continue to use it. However, in this case we should be ready to handle the exceptional cases where things doesn't go smoothly by following the accepted rules.

Currently we have very unpleasant situation with #762:
- @moodmosaic [isn't ready](https://github.com/AutoFixture/AutoFixture/pull/762#issuecomment-323075639) to review PR as this is not him expertise.
- @klimisa and @adamchester could potentially help here, but they are busy.
- @ecampidoglio, who was initially involved in this area, now is completely absent without any clue when he returns.

I've tried to reach out to @ecampidoglio during the last 1.5 months, but no luck. Also I tried to ping @klimisa (and even wrote him in twitter), but that didn't help neither. This PR is simply __dead__ for the last __2.5__ months. I wouldn't say it's very attractive to constantly pursue potential reviewers and ask them to have a look.. 😢

We are about to merge #703 that enables .NET Core support and there are people ready to volunteer in early testing. To make that convenient it would be fine to have a private NuGet feed with packages for v4. This is exactly where we need #762.

This PR is mature enough - it has 3.5 months of life so I had a lot of time to polish it up. Also I fixed all the concerns we had during the review. Now everything works out of the box - simply create a vNext branch and push it. Therefore, I want us to make a decision to merge this PR in it's current state. After all this is a _private_ feed and not a rocket science - we can always improve or rework the existing logic if it doesn't meet our needs.

@moodmosaic @adamchester @klimisa Are you fine if I merge PR #762 given all the stuff above? If not, what is your strategy?

**Comment by adamchester** [comment]
I agree #762 is an exception and I vote for moving forward with this by merging it and allowing @zvirja to improve it or revise it over time (if necessary).

I haven't reviewed it in detail, but it seems sensible to me. I think @zvirja should have the ability to do these things as he sees fit, and if he lost interest in the future, I think we should also grant the next trusted person the same rights to change it or improve it.

Ideally it would have a review and approval like everything else, but since that's not really happening, I think we should move forward.

**Comment by zvirja** [comment]
@moodmosaic Should your thumb up be considered as you agree and I can proceed? :wink:

**Comment by moodmosaic** [comment]
@zvirja It means I agree with @adamchester, so go ahead 😃

**Comment by moodmosaic** [comment]
@zvirja, perhaps you do a great job, but I currently see [7 open and 35 closed pull requests](https://github.com/AutoFixture/AutoFixture/pulls/zvirja) in a reasonable short time period: *5 months (!)*
* All these pull requests point to the right direction
* Some pull requests are small and focused
* Some other pull requests are quite big

If you *do* need my help on pull request reviews, please try to keep them small and focused. @ploeh wrote [a great blog post on this subject](http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/). Try to follow it.

For the next few months, I plan to get more busy, and for this reason, it will take me weeks before I can look at big pull requests.

**Comment by zvirja** [comment]
@moodmosaic Thank you for your time as well and you are also doing great work reviewing them!

> but I currently see 7 open and 35 closed pull requests in a reasonable short time period: 5 months (!)

Do you want to say that this is very large rate of activity and I should decrease it to produce less work for you as a reviewer? I understand that I'm producing a lot of changes, but all them should be done if we have a plan to release v4.. 

> If you do need my help on pull request reviews, please try to keep them small and focused.

Honestly, I'm a bit confused by this wish as I had a feeling that I'm already following this approach and am trying to split PRs to be as small as possible. Could you please point me to the PRs that violate these rules and your suggestions how you would do them "more appropriate"? That will help both us to align and proceed quicker in future.


> For the next few months, I plan to get more busy, and for this reason, it will take me weeks before I can look at big pull requests.

It's a pity :cry: We still have [quite a big scope](https://github.com/AutoFixture/AutoFixture/milestone/3) to do before we can release v4. However, I believe that most of the significant changes have been already done so the rest will be easier. 

Let's see how it goes. If we find that we stuck, we always can decide what to do. Probably @adamchester could also help..

**Comment by moodmosaic** [comment]
I don't think we will be stuck. After all, I've already reviewed about 3-4 PRs *opened after* my [comment](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-325541883) (!) 😃  

---

All I'm saying is *keep them coming™*, but make sure they're all small and focused. Other than @adamchester and myself, I don't see anyone else being around...

**Comment by zvirja** [comment]
Adam has a proposal [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-323188537) to allow me to apply changes to our build scripts without need to create a PR.

I track the overall picture and am trying to simplify things, so would like us to accept this rule. Additionally, I want to extend this privilege to include both build scripts and projects organization (and they both closely very related to each other). That will help me to make housekeeping simpler, while the actual business logic will not be affected. Of course, that means only insignificant changes and for significant ones (like projects removal/rename) I'll still fire PRs.

I've already spent some time on this and likely currently I'm the only one who have full and detailed vision (as @ecampidoglio is missing) about layout and processes, so that this decision should not lead to something horrible.

@moodmosaic @adamchester @klimisa Do have any objections regarding that? Remember, you will need to review less stuff 😄

**Comment by moodmosaic** [comment]
@zvirja, thank you for all the hard work you are doing.

Even though my time is limited this period, I try to follow what you're doing. You seem to be moving fast but, overall, your pull requests are all for getting v4 out the door.

For this reason, I'm fine with what you prose, but please do all these changes in pull requests, so we may go back in time and see how things evolved, if necessary.

Of course, for this kind of pull requests, you should be able to go ahead and merge them yourself, if @ecampidoglio, @adamchester, and @klimisa also agree.

**Comment by zvirja** [comment]
@moodmosaic Agree to apply changes via PRs.

@adamchester @klimisa Please put your voices as well 😉

**Comment by moodmosaic** [comment]
@zvirja, I'd suggest we wait for all (@ecampidoglio, @adamchester, and @klimisa) to agree on this, if possible.

Specially @ecampidoglio and @adamchester who have been around for years. I'm not sure if @klimisa has been active at all, to be honest (?)

**Comment by zvirja** [comment]
@moodmosaic Sure, that's exactly how I planned that.

>  Specially @ecampidoglio and @adamchester who have been around for years. 

Hope @ecampidoglio will arrive in the nearby future to reply - he was totally absent for the last months.. 

> I'm not sure if @klimisa has been active at all, to be honest (?)

That's the point that causes a bit of confusion to me. I saw very rare @klimisa's activity in PRs and this thread, however I mostly have a feeling that he doesn't participate. Given that he doesn't strong background within the project, could we consider his vote as optional? Meaning if he didn't reply - we proceed with what we have.

**Comment by klimisa** [comment]
AutoFixture has been very active the last few months, try to get to v.4 and by maintaining the existing codebase. At the moment the rate of changes happens and the time I have in order to catch up, do not align.

Presuming that more people are needed with a bit of more time than mine, I decided to leave the project. Therefore, I'm kindly asking @zvirja to withdraw my credentials from Github and from NuGet.

I will still follow the project and watch the changes but without the time pressure that it has now.

Many thanks to @pleoh and @Autofixture/Core. Special thanks to @zvirja for all the work he is doing.

**Comment by adamchester** [comment]
I agree with the [proposal here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-326537029)

Also, I didn't mean to suggest that no PR should happen, but I do of course agree that we should be able to move forward on the build changes without review if nobody has the ability or time to review them in depth.

Sorry I've been a little busy lately again, I'm trying to keep one eye on various the changes where I can.

**Comment by zvirja** [comment]
@klimisa It's a pity to hear that you don't have free time for that and decided to leave the project 😞😢  However, I do understand that decision and we indeed are doing a lot. Thank you for all the activity you did and time you spent on this project and thanks for be my colleague in the Release Management group 😄

If you have time in future, feel free to get back and contribute further to the project. Given the number of things need to be done we definitely need one more pair of hands!

@adamchester Thanks for the approval! As it has been agreed [here](#issuecomment-326542488) I'll still create PRs for better history readability, however will simply not require an approval for that.

@ecampidoglio You are the only one who left for the vote for [this](#issuecomment-326537029) proposal - @moodmosaic and @adamchester agreed. Could you please confirm that you are also fine? Hope, you will appear in the nearby future to vote so we can proceed further..

**Comment by zvirja** [comment]
@moodmosaic It seems that @ecampidoglio is still very busy to participate in this project and put his reply. Given that there is no sense of ETA when he will arrive back, may I consider the proposal above as confirmed, so I can proceed further and merge #811 (which was created about 2 weeks ago)? As I previously mentioned, I'll apply only insignificant house keeping changes and for major ones (like with the logging verbosity) I'll ask you to participate.

**Comment by moodmosaic** [comment]
@ecampidoglio, do you agree with what I proposed [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-326540892) 😃

**Comment by zvirja** [comment]
@moodmosaic It seems that your summon didn't help a lot 😄

Given that, could we proceed further as in any case we have nobody available to review those changes?

**Comment by moodmosaic** [comment]
@zvirja, it looks like @ecampidoglio [is (and will be more) busy](https://megakemp.com/2017/09/13/the-next-step/) (in a *great* way) so let's wait him for another week 😄 🎉

**Comment by zvirja** [comment]
@moodmosaic Let me speak fairly. While I do understand your intention, the options are unclear to me. Given that Enrico has been already missing for couple months and you expect that he will be even more busy, who is expected to review such PRs?

Currently you @moodmosaic and @adamchester are pretty overwhelmed with your work (and probably external life 😄). The time you can allocate for this project is hardly enough to cover the PRs with important changes. For instance, we have [this](https://github.com/AutoFixture/AutoFixture/pull/818) PR that has been opened 2 weeks ago and nobody had a chance to review it. I understand that we are not paid for the work we do so it's fine. Be sure that I respect your attention and am very thankful for that! 🏅🥇However all we should clearly understand that existing reviewers capacity _is not enough_ to cover the housekeeping.

In order to make things simpler I suggested an addition to [what we already agreed](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-323316676) - to able to modify the `csproj` and `sln` files as well. Build scripts and solution files are closely related to each other and e.g. to improve xUnit tests invocation (to get rid of [this](https://github.com/AutoFixture/AutoFixture/blob/8c6c0c670438fc3109ca3453f8e194ef8b9900ca/Build.fsx#L194) ugly list) I need to apply changes in test projects by installing an additional `xunit` packages to some projects. This changes are insignificant and they will make things simpler - all our test projects will be aligned.

It's a bit unclear to me what is special about this last rule so that 2 votes are **strongly** not enough. Previously we successfully agreed on other rules having 2 votes only.

Honestly, I'm a bit frustrated by this. I'm participating here for about half of a year now and did a lot of stuff. I have a lot of intention and readiness to move us further by actively participating in different issue and PR discussions. I think that I did enough to prove that I fully understand what I'm doing; I have clear intentions; I try analyze all the potential outcomes; I respect the spirit of the OSS project where the important changes should be discussed; I understand that my experience might be not enough so I'm always open for your opinions and I learn them. However, for me it looks like you suspect that I'll start to make a mess here if I get such permissions and don't trust me that I'll create a PR when I change something potentially significant 😞

Previously you mentioned that you are afraid that I could loose a passion and you will need to support all this by yourself. I don't see this as an issue actually. That is what already happened when Mark left this project - we had the similar situation when somebody else needed to learn how things are organized and continue to support them. That time we had me you picked that up and learned. I believe that even if I loose the passion, that will not be an issue and a new person will learn the stuff in a day or so. Actually, there less chances that I'll leave this project if I feel that I'm a part of the team and have an area I "own" and can care of 😊 

Sorry if my answer is a bit emotional, however I felt that it's better to clearly discuss that. It will allow us to move ahead or at least be aligned. And, of course, I'm happy to participate in this project and am willing to continue to do this 😉

**Comment by moodmosaic** [comment]
@zvirja, I was hoping to hear back from @ecampidoglio as he was the one who suggested that [all PRs should be approved by at least two people from the @AutoFixture/core team before being merged [..]](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-290217987).

Apparently, @ecampidoglio must be busy (which is a good thing!) as the last time he commented on this thread was [on Mar 29](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-290217987) even though he's been mentioned a couple of times after that date.

At this point, I have no idea when he's going to return, so perhaps we should move on with [this](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-328080695), as well as being able to move on quickly with the various MSBuild files you propose [here](https://github.com/AutoFixture/AutoFixture/issues/703#issuecomment-329148619).

👍 :1st_place_medal: 💯

**Comment by zvirja** [comment]
Recently I did a series of minor housekeeping changes for our code base: #846, #847, #850 and #852. The main point is that those changes doesn't affect the logic somehow and are mostly related to the code style. 

Currently I have permissions to modify our build scripts and project related files without approval. Would you agree to extend those permissions to such kind of codebase changes as well? I mean changes related to the code style (e.g. use `nameof()` for argument guards), when actual logic remains the same. As usual - all changes will be done via PRs and if there are significant changes - I'll ask for the review.

@moodmosaic @adamchester @ecampidoglio Do you agree with that? That should simplify the amount of PRs that require the review and allow to focus on valuable changes only.

**Comment by moodmosaic** [comment]
@zvirja, fine by me 👍 Thank you for all the hard work you are doing.

**Comment by adamchester** [comment]
@zvirja find by me too :)

**Comment by zvirja** [comment]
Thanks for the confirmation :sunglasses:

Given that Enrico is completely absent and doesn't participate for now, I'm considering this rule as agreed. @moodmosaic @adamchester Let me know if you disagree.

**Comment by zvirja** [comment]
Previously we have agreed on [a rule](https://github.com/AutoFixture/AutoFixture/wiki/Project-processes#pull-request-review) that PR can be merged if it has one approval and one week of lifetime. The reason behind this week was to give an opportunity for other reviewers to jump-in. However, since we agreed on this rule, in fact that never happened (in more than 2 months). The reality is that @moodmosaic and me are the only alive maintainers of this project at this moment of time. Therefore, this one week of delay doesn't make any sense.

Given that I have a proposal to eliminate that one week delay rule, so that PRs can be merged immediately if they has one approval. Of course, we still left the possibility for reviewers to override the rule per PR, however by default PRs will be "mergable" immediately.

@adamchester @moodmosaic What are you thoughts regarding that?

**Comment by moodmosaic** [comment]
>so that PRs can be merged immediately if they have one approval

👍 Fine by me.

**Comment by adamchester** [comment]
Agreed, yes 👍

**Comment by moodmosaic** [comment]
This [comment](https://github.com/AutoFixture/AutoFixture/issues/708#issuecomment-371460131) by @ploeh helped me realize that I'm here, helping with issues, as long as the overall vision remains quite the same―I'll be pretty much useless if there's a *different* vision and so then I'll be out...

**Comment by zvirja** [comment]
@moodmosaic I've described my vision in [my reply](https://github.com/AutoFixture/AutoFixture/issues/708#issuecomment-371625833). As far as I can see, I still have the pretty same vision as Mark, nothing completely different. I feel that your help and experience are very valuable for the project and I'd be very happy to still see you here. It's totally fine that sometimes we might have a different vision on some things - that allows to make the best decisions.
